kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/data/offer/OfferRepository.java

```diff
@@ -43,6 +48,17 @@ public static OfferRepository getInstance() {
         return instance;
     }
 
+    private void listenToPendingOrders() {
+        orderRepository.addOrderObserver(new Observer<Order>() {
+            @Override
+            public void onChanged(Order order) {
+                if(order.getStatus() == Status.PENDING) {
+                    removeFromCachedOfferList(order.getOfferId());
```

Comment:
do we need any action  if order.getStatus() == Status.FAILED?

Reply:
lets talk